src/main/scala/za/co/absa/commons/spark/SchemaUtils.scala

```diff
@@ -0,0 +1,214 @@
+    }
+  }
+
+  /**
+   * Finds all differences of two StructFields and returns their paths
+   *
+   * @param field1 The first field to compare
+   * @param field2 The second field to compare
+   * @param parent Parent path. This is used for the accumulation of differences and their print out
+   * @return Returns a Seq of found difference paths in scheme in the StructField
+   */
+  private def diffField(field1: StructField, field2: StructField, parent: String): Seq[String] = {
+    field1.dataType match {
+      case _ if field1.dataType.typeName != field2.dataType.typeName =>
+        Seq(s"$parent.${field1.name} data type doesn't match (${field1.dataType.typeName}) vs (${field2.dataType.typeName})")
+      case arrayType1: ArrayType =>
+        diffArray(arrayType1, field2.dataType.asInstanceOf[ArrayType], s"$parent.${field1.name}")
+      case structType1: StructType =>
+        diffSchema(structType1, field2.dataType.asInstanceOf[StructType], s"$parent.${field1.name}")
+      case _ =>
+        Seq.empty[String]
+    }
+  }
+
+  /**
+   * Returns data selector that can be used to align schema of a data frame. You can use [[alignSchema]].
+   *
+   * @param schema Schema that serves as the model of column order
+   * @return Sorted DF to conform to schema
+   */
+  def getDataFrameSelector(schema: StructType): List[Column] = {
+    import za.co.absa.spark.hofs._
+
+    def processArray(arrType: ArrayType, column: Column, name: String): Column = {
+      arrType.elementType match {
+        case arrType: ArrayType =>
+          transform(column, x => processArray(arrType, x, name)).as(name)
+        case nestedStructType: StructType =>
+          transform(column, x => struct(processStruct(nestedStructType, Some(x)): _*)).as(name)
+        case _ => column
+      }
+    }
+
+    def processStruct(curSchema: StructType, parent: Option[Column]): List[Column] = {
+      curSchema.foldRight(List.empty[Column])((field, acc) => {
+        val currentCol: Column = parent match {
+          case Some(x) => x.getField(field.name).as(field.name)
+          case None    => col(field.name)
+        }
+        field.dataType match {
+          case arrType: ArrayType     => processArray(arrType, currentCol, field.name) :: acc
+          case structType: StructType => struct(processStruct(structType, Some(currentCol)): _*).as(field.name) :: acc
+          case _                      =>  currentCol :: acc
+        }
+      })
+    }
+
+    processStruct(schema, None)
+  }
+
+  /**
+   * Using schema selector from [[getDataFrameSelector]] aligns the schema of a DataFrame to the selector for operations
+   * where schema order might be important (e.g. hashing the whole rows and using except)
+   *
+   * @param df DataFrame to have it's schema aligned/sorted
+   * @param selector Selector for the DataFrame
+   * @return Returns aligned and filtered schema
+   */
+  def alignSchema(df: DataFrame, selector: List[Column]): DataFrame = df.select(selector: _*)
```

Comment:
If this method is intended to be used in combination with `getDataFrameSelector` I'd suggest to make this dependency more explicit and reflect it in the method signature to avoid misuse of misread.
For example, one option I would consider is to change the signature to `(DataFrame, StructType) and just call `getDataFrameSelector` from `alignSchema`. Another option would be to introduce a type that represents the mentioned selector if you want to keep both methods independent.